searx/plugins/__init__.py

```diff
@@ -77,6 +87,81 @@ def call(self, ordered_plugin_list, plugin_type, request, *args, **kwargs):
         return ret
 
 
+def load_external_plugins(plugin_names):
+    plugins = []
+    for name in plugin_names:
+        logger.debug('loading plugin: {0}'.format(name))
+        try:
+            pkg = import_module(name)
+        except Exception as e:
+            logger.critical('failed to load plugin module {0}: {1}'.format(name, e))
+            exit(3)
+
+        pkg.__base_path = dirname(abspath(pkg.__file__))
+
+        fix_package_resources(pkg, name)
+
+        plugins.append(pkg)
+        logger.debug('plugin "{0}" loaded'.format(name))
+    return plugins
+
+
+def check_resource(base_path, resource_path, name, target_dir, plugin_dir):
+        dep_path = join(base_path, resource_path)
+        file_name = basename(dep_path)
+        resource_path = join(target_dir, file_name)
+        if not exists(resource_path) or sha_sum(dep_path) != sha_sum(resource_path):
```

Comment:
Sorry for  question, but what is the intention of the sha_sum .. performance? I mean, reading both files completely into the memory to see that the sha is different and than copy the file from the resource to the destination (which includes another read) is this really faster?  .. wouldn't it be better to just compare the timestamps on the _right_ and _left_ hand?

Reply:
This happens only once at startup, don't think that the performance difference really matters, but checking timestamps is another good way to achieve similar functionality